cont.c

```diff
@@ -2237,6 +2265,117 @@ rb_fiber_terminate(rb_fiber_t *fiber, int need_interrupt)
     fiber_switch(next_fiber, 1, &value, RB_NO_KEYWORDS, Qfalse, false);
 }
 
+static void
+fiber_check_canceled()
+{
+    rb_fiber_t *fiber = fiber_current();
+    VALUE reason = fiber->cancel_reason;
+    if (fiber->canceling && reason != Qundef) {
+        VM_ASSERT(reason  != Qundef);
```

Comment:
This assertion feels too paranoid, doesn't it?

Reply:
Haha, yes. it got left in from an earlier revision. :)